samtranslator/validator/validator.py

```diff
@@ -7,49 +9,254 @@
 
 
 class SamTemplateValidator(object):
+    """
+    SAM template validator
+    """
+
+    # Useful to find a unicode prefixed string type to replace it with the non unicode version
+    # On Python2, the validator returns types in error messages prefixed with 'u'
+    # We remove them to be consistent between python versions
+    # Example: "u'integer'" -> "'integer'"
+    UNICODE_TYPE_REGEX = re.compile("u('[^']+')")
+
+    def __init__(self, schema=None):
+        """
+        Constructor
+
+        Parameters
+        ----------
+        schema_path : str, optional
+            Path to a schema to use for validation, by default None, the default schema.json will be used
+        """
+        # Replaces the static method with the instance method
+        # Otherwise the instance would call the static one
+        # Static one is kept for backward compatibility
+        self.validate = self._validate
+
+        if not schema:
+            schema = self._read_json(sam_schema.SCHEMA_NEW_FILE)
+
+        # Helps resolve the $Ref to external files
+        # schema content must be passed to resolve local (#/def...) references
+        resolver = jsonschema.RefResolver("file://" + sam_schema.SCHEMA_DIR + "/", schema)
```

Comment:
Will this work on windows? I see a "/", which might cause problems for SAM CLI?

Reply:
For URIs, "/" works on Windows, but we should verify it nonetheless.